pkg/controller/route53resolver/associateresolverrule/hooks.go

```diff
@@ -0,0 +1,162 @@
+/*
+Copyright 2021 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package associateresolverrule
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"k8s.io/client-go/util/workqueue"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	"github.com/crossplane/crossplane-runtime/pkg/event"
+	"github.com/crossplane/crossplane-runtime/pkg/logging"
+	"github.com/crossplane/crossplane-runtime/pkg/meta"
+	"github.com/crossplane/crossplane-runtime/pkg/ratelimiter"
+	"github.com/crossplane/crossplane-runtime/pkg/reconciler/managed"
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+
+	"github.com/crossplane/provider-aws/apis/route53resolver/manualv1alpha1"
+	"github.com/crossplane/provider-aws/pkg/clients/associateresolverrule"
+
+	awsclient "github.com/crossplane/provider-aws/pkg/clients"
+)
+
+const (
+	errUnexpectedObject = "The managed resource is not an AssociatedResolverRule resource"
+	errCreate           = "failed to create the AssociatedResolverRule"
+	errDelete           = "failed to delete the AssociatedResolverRule"
+	errGet              = "failed to get the AssociatedResolverRule"
+)
+
+// SetupAssociateResolverRule adds a controller that reconciles AssociateResolverRule
+func SetupAssociateResolverRule(mgr ctrl.Manager, l logging.Logger, rl workqueue.RateLimiter, poll time.Duration) error {
+	name := managed.ControllerName(manualv1alpha1.AssociateResolverRuleGroupKind)
+
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		WithOptions(controller.Options{
+			RateLimiter: ratelimiter.NewController(rl),
+		}).
+		For(&manualv1alpha1.AssociateResolverRule{}).
+		Complete(managed.NewReconciler(mgr,
+			resource.ManagedKind(manualv1alpha1.AssociateResolverRuleGroupVersionKind),
+			managed.WithExternalConnecter(&connector{kube: mgr.GetClient(), newRoute53ResolverClientFn: associateresolverrule.NewRoute53ResolverClient}),
+			managed.WithInitializers(managed.NewDefaultProviderConfig(mgr.GetClient()), managed.NewNameAsExternalName(mgr.GetClient())),
```

Comment:
```suggestion
			managed.WithInitializers(),
```
Its external name is computed by AWS.